provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -544,7 +576,10 @@ func generateSchema() schema.PackageSpec {
 							"https://www.pulumi.com/docs/intro/concepts/programming-model/#autonaming",
 					},
 					"proxy": {
-						TypeSpec: schema.TypeSpec{Type: "string"},
+						TypeSpec: schema.TypeSpec{
+							Type:  "string",
+							Plain: true,
+						},
```

Comment:
This is plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/cluster.ts#L1257-L1275